app/services/licences/process-supplementary-billing-flag.service.js

```diff
@@ -0,0 +1,78 @@
+'use strict'
+
+/**
+ * Orchestrates flagging a licence for supplementary billing
+ * @module ProcessSupplementaryBillingFlagService
+ */
+
+const ChargeVersionModel = require('../../models/charge-version.model.js')
+const FlagSupplementaryBillingService = require('./flag-supplementary-billing.service.js')
+const DetermineSupplementaryBillingYearsService = require('./determine-supplementary-billing-years.service.js')
+
+/**
+ * Orchestrates flagging a licence for supplementary billing based on the provided charge version.
+ *
+ * If a `chargeVersionId` is present in the payload, the service fetches the charge version, along with its relevant
+ * charge references and licence details.
+ *
+ * If the charge version has any two-part tariff indicators on any of its charge references then it calls the
+ * `DetermineSupplementaryBillingYearsService` where it will work out which financial year ends have been affected by
+ * the change in the charge version.
+ *
+ * If there are years that have been affected, these are then passed to our `FlagSupplementaryBillingService`, which
+ * checks if a bill run has already been sent for that affected year, and if it has it will then persist the flag in the
+ * `LicenceSupplementaryYear` table.
+ *
+ * @param {Object} payload - The payload from the request to be validated
+ */
+async function go (payload) {
+  try {
+    if (!payload.chargeVersionId) {
+      return
+    }
+
+    const { chargeReferences, licence, endDate, startDate } = await _fetchChargeVersion(payload.chargeVersionId)
+    const twoPartTariffIndicator = _twoPartTariffIndicators(chargeReferences)
+
+    if (!twoPartTariffIndicator) {
+      return
+    }
+
+    const years = await DetermineSupplementaryBillingYearsService.go(startDate, endDate)
+
+    if (years) {
+      await FlagSupplementaryBillingService.go(licence, years)
+    }
```

Comment:
I think to be consistent with the 2 guard clauses above, we could tweak this section.

```suggestion
    if (!years) {
      return
    }

    await FlagSupplementaryBillingService.go(licence, years)
```